### PR TITLE
Tracing: traces now correctly form a tree

### PR DESCRIPTION
**Pull request by crufter**
To support https://github.com/micro/platform/pull/14

### CODE REVIEW

**Comment by asim** [comment]
Self merge when you are ready!

File: debug/trace/memory/memory.go [hunk 1]
```diff
@@ -85,6 +85,6 @@ func NewTracer(opts ...trace.Option) trace.Tracer {
 	return &Tracer{
 		opts: options,
 		// the last 64 requests
-		buffer: ring.New(64),
+		buffer: ring.New(640),
```
Comment (crufter) [inline debug/trace/memory/memory.go]: This might be a bit controversial. Also need to fix comments.

Comment (asim) [reply debug/trace/memory/memory.go]: That is definitely controversial. Max I'd do is 256

File: debug/trace/trace.go [hunk 2]
```diff
@@ -34,17 +35,32 @@ type Span struct {
...
+	parentSpanID, ok := metadata.Get(ctx, spanIDKey)
+	return traceID, parentSpanID, ok
 }
 
-// NewContext creates a new context with the span
-func NewContext(ctx context.Context, s *Span) context.Context {
-	return context.WithValue(ctx, spanKey{}, s)
+// ToContext saves the trace and span ids in the context
+func ToContext(ctx context.Context, traceID, parentSpanID string) context.Context {
+	return metadata.MergeContext(ctx, map[string]string{
```
Comment (jakexks) [inline debug/trace/trace.go:60]: Does this work if ctx is nil?

Comment (crufter) [reply debug/trace/trace.go:60]: Good catch.

**Pull request merged**